routers/api/v1/api.go

```diff
@@ -771,6 +771,14 @@ func RegisterRoutes(m *macaron.Macaron) {
 						m.Delete("", bind(api.DeleteFileOptions{}), repo.DeleteFile)
 					}, reqRepoWriter(models.UnitTypeCode), reqToken())
 				}, reqRepoReader(models.UnitTypeCode))
+				m.Group("/topics", func() {
+					m.Combo("").Get(repo.ListTopics).
+						Put(reqToken(), reqRepoWriter(models.UnitTypeCode), bind(api.RepoTopicOptions{}), repo.UpdateTopics)
+					m.Group("/:topic", func() {
+						m.Combo("").Put(reqToken(), reqRepoWriter(models.UnitTypeCode), repo.AddTopic).
```

Comment:
Only admin of the repository could change the topics?

Reply:
I actually thought write access could change topics, but I see now admin is required. I have corrected this, and also added additional access test.